packages/knip/src/reporters/codeclimate.ts

```diff
@@ -0,0 +1,127 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { toRelative } from '../util/path.js';
+import { getTitle } from './util.js';
+
+type CodeClimateSeverity = 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+
+interface CodeClimateEntry {
+  type: 'issue';
+  check_name: string;
+  description: string;
+  categories: string[];
+  location: {
+    path: string;
+    positions: {
+      begin: {
+        line: number;
+        column: number;
+      };
+    };
+  };
+  severity: CodeClimateSeverity;
+  fingerprint: string;
+}
+
+export default async ({ report, issues }: ReporterOptions) => {
+  const entries: CodeClimateEntry[] = [];
+
+  for (const [type, isReportType] of Object.entries(report) as Entries<Report>) {
+    if (!isReportType) {
+      continue;
+    }
+
+    if (type === 'files') {
+      continue;
```

Comment:
Just noticed that the tests don't show unused files, while there should be a few instances of those. So I did some work in `main` to make the `_files` issue type consistent with the others.

Any chance you could rebase from `main` and do [something like this](https://github.com/webpro-nl/knip/blob/171c86a392bb4f0046e7e9e71a9f311fbc8ce309/packages/knip/src/reporters/symbols.ts#L56) so we'll also have unused files in the report?